service/matching/taskQueueManager.go

```diff
@@ -77,7 +82,7 @@ type (
 	}
 
 	taskQueueManager interface {
-		Start()
+		Start() error
```

Comment:
what is our strategy? start can return error?

Reply:
Yes.

Reply:
let us wait for the DI framework, if DI allows / can handle start & stop returning error, then it should be fine, otherwise this will cause caller to handle errors everywhere

Reply:
DI won't call start/stop.

Reply:
i do not believe we have talked about the `allow or disallow start / stop`?

depends on the DI, at one of the DI option can do start / stop
https://github.com/uber-go/fx/blob/ce8d4f0bff6c78f22aabd6bc33fbd56f255cc490/example_test.go#L146